frontend/ui/widget/statuswidget.lua

```diff
@@ -0,0 +1,653 @@
+        }
+        local title_text_container = CenterContainer:new {
+            dimen = Geom:new { w = width, h = text_title:getSize().h },
+            text_title
+        }
+        table.insert(title_author_progressbar_group, title_text_container)
+    end
+
+    local text_author = TextWidget:new {
+        text = authors,
+        face = self.small_font_face,
+        padding = 2,
+    }
+
+    local author_container = CenterContainer:new {
+        dimen = Geom:new { w = width, h = text_author:getSize().h },
+        text_author
+    }
+
+    table.insert(title_author_progressbar_group, author_container)
+
+    local progressWidget = ProgressWidget:new {
+        width = width * 0.7,
+        height = Screen:scaleBySize(10),
+        percentage = current_page / total_pages,
+        ticks = {},
+        tick_width = 0,
+        last = total_pages,
+    }
+
+    local progress_bar_container = CenterContainer:new {
+        dimen = Geom:new { w = width, h = progressWidget:getSize().h },
+        progressWidget
+    }
+
+    table.insert(title_author_progressbar_group, progress_bar_container)
+    local text_complete = TextWidget:new {
+        text = string.format("%1.f", progressWidget.percentage * 100) .. "% " .. _("Completed"),
+        face = self.small_font_face,
+    }
+
+    local progress_bar_text_container = CenterContainer:new {
+        dimen = Geom:new { w = width, h = text_complete:getSize().h },
+        text_complete
+    }
+
+    table.insert(title_author_progressbar_group, progress_bar_text_container)
+    table.insert(title_author_container, title_author_progressbar_group)
+    return title_author_container
+end
+
+
+function StatusWidget:onAnyKeyPressed()
+    return self:onClose()
+end
+
+function StatusWidget:onClose()
+    self:saveSummary()
+    UIManager:setDirty("all")
+    UIManager:close(self)
+    return true
+end
+
+
+--TODO: MOVE TO UTILS AND CHANGE ALSO TEXTBOXWIDGET
+function StatusWidget:_wrapGreedyAlg(h_list, width)
+    local cur_line_width = 0
+    local cur_line = {}
+    local v_list = {}
+
+    for k, w in ipairs(h_list) do
+        w.box = {
+            x = cur_line_width,
+            w = w.width,
+        }
+        cur_line_width = cur_line_width + w.width
+        if w.word == "\n" then
+            if cur_line_width > 0 then
+                -- hard line break
+                table.insert(v_list, cur_line)
+                cur_line = {}
+                cur_line_width = 0
+            end
+        elseif cur_line_width > width then
+            -- wrap to next line
+            table.insert(v_list, cur_line)
+            cur_line = {}
+            cur_line_width = w.width
+            table.insert(cur_line, w)
+        else
+            table.insert(cur_line, w)
+        end
+    end
+    -- handle last line
+    table.insert(v_list, cur_line)
+
+    return v_list
+end
+
+--TODO: MOVE TO UTILS AND CHANGE ALSO TEXTBOXWIDGET
+function StatusWidget:_getVerticalList(text, width, face, bold)
```

Comment:
I assume you need this to wrap long book titles? I also assume you already tried `TextBoxWidget` since that's where the code comes from. Why is `TextBoxWidget` not working? We should probably fix that instead ;P